### PR TITLE
Add maintenance4 to GpuValidationArrayOOBGraphicsShaders

### PR DESCRIPTION
**Pull request by Tony-LunarG**
Works around shader input/output mismatches

### CODE REVIEW

File: tests/vklayertests_gpu.cpp [hunk 1]
```diff
@@ -57,14 +58,16 @@ TEST_F(VkGpuAssistedLayerTest, GpuValidationArrayOOBGraphicsShaders) {
...
+    auto features2 = LvlInitStruct<VkPhysicalDeviceFeatures2KHR>(&maintenance4_features);
     auto indexing_features = LvlInitStruct<VkPhysicalDeviceDescriptorIndexingFeaturesEXT>();
     if (descriptor_indexing) {
         PFN_vkGetPhysicalDeviceFeatures2KHR vkGetPhysicalDeviceFeatures2KHR =
             (PFN_vkGetPhysicalDeviceFeatures2KHR)vk::GetInstanceProcAddr(instance(), "vkGetPhysicalDeviceFeatures2KHR");
         ASSERT_TRUE(vkGetPhysicalDeviceFeatures2KHR != nullptr);
 
-        features2 = LvlInitStruct<VkPhysicalDeviceFeatures2KHR>(&indexing_features);
+        maintenance4_features.pNext = &indexing_features;
         vkGetPhysicalDeviceFeatures2KHR(gpu(), &features2);
```
Comment (ncesario-lunarg) [inline tests/vklayertests_gpu.cpp:71]: You can use `GetPhysicalDeviceFeaures2(maintenance_features);` if you want to get rid of the function pointer in the test.

File: tests/vklayertests_gpu.cpp [hunk 2]
```diff
@@ -41,6 +41,7 @@ TEST_F(VkGpuAssistedLayerTest, GpuValidationArrayOOBGraphicsShaders) {
         "GPU validation: Verify detection of out-of-bounds descriptor array indexing and use of uninitialized descriptors.");
 
     SetTargetApiVersion(VK_API_VERSION_1_1);
+    AddRequiredExtensions(VK_KHR_MAINTENANCE_4_EXTENSION_NAME);
```
Comment (ncesario-lunarg) [inline tests/vklayertests_gpu.cpp:44]: Is maintenance4 optional? If so, might be good to use the `AddOptionalExtension`. If it's required, does it make sense to move the `AreRequiredExtensionsEnabled()` check to right after the `InitGpuAssistedFramework` call?

**Review by ncesario-lunarg — APPROVED** [review]
Just some testing framework questions (since that stuff is fresh in my mind for some reason :)), but LGTM!

File: tests/vklayertests_gpu.cpp (hunk 2 above)
Comment (Tony-LunarG) [reply tests/vklayertests_gpu.cpp:44]: Moved the check.  I suppose I could only run the geometry shaders if it's present, but it's common enough that I think our coverage is still good if we require it.

**Review by jeremyg-lunarg — APPROVED** [review]

**Pull request merged**